task07-consume-real-data/index.js

```diff
@@ -0,0 +1,143 @@
+document.addEventListener("DOMContentLoaded", function() {
+    paintData();    
+});
+
+function readMoreOrLess(){
+    let readMoreDiv = document.getElementById("readMoreSipnosis")
+    let readMoretext = document.getElementById("readMoreText")
+    let readMoreIcon = document.getElementById("readMoreIcon")
+    let logoReadMore = document.getElementById("logoReadMore")
+    
+    if(window.getComputedStyle(readMoreDiv).display == 'none'){
+        logoReadMore.style.display = 'block';
+        readMoreDiv.style.display = 'block';
+        readMoretext.innerText = "Read Less"
+        readMoreIcon.src ="images/readLess.svg"
+
+    }
+    else{
+        logoReadMore.style.display = 'none';
+        readMoreDiv.style.display = 'none';
+        readMoretext.innerText = "Read All"
+        readMoreIcon.src ="images/readMore.svg"
+    }
+    
+}
+
+function showOrHideMenu(){
+
+    let menuOptions = document.getElementById("menuOptions")
+    let iconMenu = document.getElementById("iconMenu")
+    if(window.getComputedStyle(menuOptions).display == 'none'){
+        menuOptions.style.display = 'block';
+        iconMenu.src ="images/close.svg"
+    }
+    else{
+        menuOptions.style.display = 'none';
+        iconMenu.src ="images/menu.svg"
+    }
+}
+async function readRealData() {
+    try {
+        let response =  await fetch("https://static.akurey.com/trainings/OnePieceInformation.json");
+        if (!response.ok) {
+        throw new Error(`HTTP error: ${response.status}`);
+        }
+        const json = await response.json();
+        return json;
+    }
+    catch(error) {
+        console.error(`Could not get data: ${error}`);
+    }
+}
+function compareStrings(a, b) {
+    return (a < b) ? -1 : (a > b) ? 1 : 0;
+}
+
+const paintData = ()=>{
+    const json = readRealData();
+    json.then((result)=>{
+        let sortedIslands = result.islands.sort(function(a, b) {
```

Comment:
in arrow function the sort would look like `.sort((a,b)=>compareStrings(a.name, b.name))` it looks way cleaner and concise

Reply:
Done!!